### PR TITLE
Bump openshift/api

### PR DESCRIPTION
**Pull request by pierreprinetti**
This bump is instrumental to the development of OpenStack FailureDomains.

### CODE REVIEW

**Comment by yuqi-zhang** [comment]
The current api bump is being performed via https://github.com/openshift/machine-config-operator/pull/3269

Let's rebase on that as necessary

/hold

**Comment by sinnykumari** [comment]
openshift/api bump has been done with PR https://github.com/openshift/machine-config-operator/pull/3269 merged. If you still need further api bump, please rebase this PR on top of master to get required changes

**Comment by pierreprinetti** [comment]
@yuqi-zhang I am not sure what to do with that failing `verify` test. Would you know?

**Comment by pierreprinetti** [comment]
@yuqi-zhang passed `make verify` by bumping `golangci-lint`. PTAL!

File: go.mod [hunk 1]
```diff
@@ -21,41 +21,41 @@ require (
...
 	github.com/golang/glog v1.0.0
-	github.com/golangci/golangci-lint v1.46.2
+	github.com/golangci/golangci-lint v1.49.0
 	github.com/google/go-cmp v0.5.8
 	github.com/google/renameio v0.1.0
 	github.com/imdario/mergo v0.3.13
 	github.com/opencontainers/go-digest v1.0.0
-	github.com/openshift/api v0.0.0-20220824134416-0f86a223d4bc
+	github.com/openshift/api v0.0.0-20220915134421-1265e9925688
 	github.com/openshift/client-go v0.0.0-20220525160904-9e1acff93e4a
```
Comment (sinnykumari) [inline go.mod:30]: shouldn't we also bump other openshift libs as well like client-go, library-go  to latest?

Comment (pierreprinetti) [reply go.mod:30]: Fine by me, will do.

**Review by yuqi-zhang — COMMENTED** [review]
Hey, sorry I didn't get to this earlier. It looks like the kubevirt platform type was updated, namely this: 
https://github.com/openshift/machine-config-operator/blob/master/pkg/controller/template/render_test.go#L86

Should be `external` now, and it should pass

Otherwise, generally seems fine

**Review by sinnykumari — APPROVED** [review]
LGTM, Thanks for the PR

**Comment by sinnykumari** [comment]
Since https://github.com/openshift/machine-config-operator/pull/3269 has been merged, hold is not needed
/hold cancel

**Comment by pierreprinetti** [comment]
Openstack tests are currently disrupted, e2e-openstack is not expected to succeed on 4.12.

**Comment by yuqi-zhang** [comment]
hmm, the status is outdated, just to check

/test bootstrap-unit

**Comment by yuqi-zhang** [comment]
Ok, looks like unit tests are good now. Do you expect this to affect any specific platform? Should we run some additional tests (e.g. payload tests) to make sure?

And also to check, we expect e2e-openstack to fail right?

**Review by yuqi-zhang — APPROVED** [review]
Thanks for the context. I think we have enough time to fix any issues. Since this is only a bump, we should be ok

/lgtm

**Pull request merged**